Sources/PointFree/Admin/Index.swift

```diff
@@ -0,0 +1,138 @@
+import Css
+import Either
+import Foundation
+import Html
+import HtmlCssSupport
+import HttpPipeline
+import HttpPipelineHtmlSupport
+import Optics
+import Prelude
+import Styleguide
+import Tuple
+
+private let adminEmails = [
+  "mbw234@gmail.com",
+  "stephen.celis@gmail.com"
+]
+
+func requireAdmin<A>(
+  _ middleware: @escaping Middleware<StatusLineOpen, ResponseEnded, T2<Database.User, A>, Data>
+  ) -> Middleware<StatusLineOpen, ResponseEnded, T2<Database.User, A>, Data> {
+
+  return { conn in
+    conn
+      |> (adminEmails.contains(get1(conn.data).email.unwrap) ? middleware : redirect(to: .secretHome))
+  }
+}
+
+let adminIndex =
+  requireUser
+    <<< requireAdmin
```

Comment:
stacking these transformers makes sure that the user is logged in and is an admin